File: app/src/main/java/com/maciejkozlowski/databases/Timings.java
package com.maciejkozlowski.databases;

import android.util.Log;

/**
 * Created by Maciej Kozłowski on 04.05.17.
 */
public class Timings {

    private String tag;

    private long start;

    public Timings(String tag) {
        this.tag = tag;
    }

    public void start() {
        start = System.currentTimeMillis();
    }

    public void logTime(String label) {
        long time = System.currentTimeMillis() - start;
        log(label + ": " + time);
    }

    public void log(String label) {
        Log.d(tag, label);
    }
}

File: app/src/main/java/com/maciejkozlowski/databases/Timings.kt
package com.maciejkozlowski.databases

import android.util.Log

import com.maciejkozlowski.databases.results.Result

/**
 * Created by Maciej Kozłowski on 04.05.17.
 */
class Timings(private val tag: String) {

    private var start = 0L

    fun start() {
        start = System.currentTimeMillis()
    }

    fun logTime(result: Result, label: String, size: Int) {
        val time = System.currentTimeMillis() - start
        //log("$label: $time")
        result.addTime("$size-$tag", time)
    }

    private fun log(label: String) {
        Log.d("###$tag", label)
    }
}
